nomos-services/blend/src/core/mod.rs

```diff
@@ -92,92 +91,120 @@ where
-        let mut cryptographic_processor = CryptographicProcessor::new(
-            blend_config.crypto.clone(),
-            membership.clone(),
-            rng.clone(),
-        );
-        let network_relay = overwatch_handle.relay::<NetworkService<_, _>>().await?;
-        let network_adapter = Network::new(network_relay);
 
-        // Incoming streams
+        let blend_config = service_resources_handle
+            .settings_handle
+            .notifier()
+            .get_updated_settings();
+        let membership = blend_config.membership();
+        let minimum_network_size = blend_config.minimum_network_size;
 
-        // Yields once every randomly-scheduled release round.
-        let mut message_scheduler = UninitializedMessageScheduler::<
-            _,
-            _,
-            ProcessedMessage<Network::BroadcastSettings>,
-        >::new(
-            blend_config.session_stream(),
-            blend_config.scheduler_settings(),
-            rng.clone(),
-        )
-        .wait_next_session_start()
-        .await;
+        // TODO: Add logic to try process new sessions. I.e:
+        // * If the old session membership was too small and the new one is large
+        //   enough, create a new backend and start receiving messages to blend on the
+        //   inbound channel.
+        // * If the old and new session membership are both too small, do nothing.
+        // * If the old session membership was large and the new one too small, perform
+        //   the session rotation on the backend but stop listening on the inbound
+        //   channel, since we don't want to blend new messages.
+        // * If the old and new session membership are both large enough, perform
+        //   session rotation and keep listening on incoming messages.
+        // Ideally, this service would be stopped altogether by the proxy service when a
+        // session is too small. Yet, the service itself must be resilient in case of
+        // bugs where the proxy service does not do that, hence the need for this
+        // additional logic.
+        if membership.size() < minimum_network_size.get() as usize {
+            tracing::warn!(target: LOG_TARGET, "Blend network size is smaller than the required minimum. Not starting swarm, hence no messages will be blended nor any Blend message will be received in this session.");
+            // We still mark the service as ready, albeit other services won't be able to
+            // interact with this service by sending messages to it, and it indeed should
+            // not happen, as all interactions should happen via the proxy service.
+            service_resources_handle.status_updater.notify_ready();
+            tracing::info!(
+                target: LOG_TARGET,
+                "Service '{}' is ready.",
+                <RuntimeServiceId as AsServiceId<Self>>::SERVICE_ID
+            );
+        } else {
+            let mut rng = ChaCha12Rng::from_entropy();
```

Comment:
Chacha12 should not be used anymore. Any rng should use the blake based one as per spec iirc